common/submit.py

```diff
@@ -33,6 +38,11 @@ def store_submit(request: HttpRequest, assignment: AssignedTask) -> Submit:
     Optionally, the request can also contain a "paths" field that contains a string with a single
     path per line, assigning paths to the uploaded "solution" files.
     """
+    if (
```

Comment:
This check has to be modified (see other comment), in case deadline is `NULL`, but `hard_deadline` is set.